### PR TITLE
Replace lodash's each/map with ES5 native forEach/map

### PR DESCRIPTION
**Pull request by astorije**
And use some fat arrows because these are the good kind of fats.

### CODE REVIEW

**Comment by MaxLeiter** [comment]
Maybe worth noting that according to [this ](https://stackoverflow.com/questions/18881487/why-is-lodash-each-faster-than-native-foreach#18884620), lodash's forEach is faster than the native implementation

**Comment by astorije** [comment]
It's slightly more complicated than that: it will be faster on *some* browsers and similar in others because of Lodash's implementation and how browsers implement `forEach`.

That being said, while it's good to keep in mind in case this ever becomes a bottleneck, I don't think we should worry about that. While we're at it, why not using `for` and `while` loops :)

Thanks for bringing that up though, I didn't know that before you posted it.

**Comment by xPaw** [comment]
Keep in mind we only use lodash on the server, so the nide is the only engine. :+1: for using native methods when possible.

File: src/plugins/irc-events/names.js [hunk 1]
```diff
@@ -13,11 +12,7 @@ module.exports = function(irc, network) {
 
 		chan.users = [];
 
-		_.each(data.users, function(u) {
-			var user = new User(u, network.prefixLookup);
-
-			chan.users.push(user);
-		});
+		data.users.forEach(u => chan.users.push(new User(u, network.prefixLookup)));
```
Comment (xPaw) [inline src/plugins/irc-events/names.js]: `map` should work fine here

**Comment by astorije** [comment]
Re: micro-change performance, on that note I watched a nice talk about how JSPerf-based conclusions are often not applicable :) https://www.youtube.com/watch?v=65-RbBwZQdU

**Pull request merged**